invenio_communities/assets/semantic-ui/js/invenio_communities/settings/curation-policy.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2023 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import { i18next } from "@translations/invenio_communities/i18next";
+import { Formik } from "formik";
+import _defaultsDeep from "lodash/defaultsDeep";
+import React, { Component } from "react";
+import ReactDOM from "react-dom";
+import { FieldLabel, RichInputField, CKEditorConfig } from "react-invenio-forms";
+import { Button, Form, Grid, Icon, Message } from "semantic-ui-react";
+import { CommunityApi } from "../api";
+import { communityErrorSerializer } from "../api/serializers";
+import PropTypes from "prop-types";
+import * as Yup from "yup";
+
+const COMMUNITY_VALIDATION_SCHEMA = Yup.object({
+  metadata: Yup.object({
+    curation_policy: Yup.string().max(2000, "Maximum number of characters is 2000"),
+  }),
+});
+
+class CommunityCurationPolicyForm extends Component {
+  state = {
+    error: "",
+    isSaved: false,
+  };
+  getInitialValues = () => {
+    const { community } = this.props;
+    let initialValues = _defaultsDeep(community, {
+      metadata: {
+        curation_policy: "",
+      },
+    });
+
+    return initialValues;
+  };
+
+  setGlobalError = (errorMsg) => {
+    this.setState({ error: errorMsg });
+  };
+
+  setIsSavedState = (newValue) => {
+    this.setState({ isSaved: newValue });
+  };
+
+  onSubmit = async (values, { setSubmitting, setFieldError }) => {
+    const { community } = this.props;
+    setSubmitting(true);
+
+    try {
+      const client = new CommunityApi();
+      await client.update(community.id, values);
+
+      this.setIsSavedState(true);
+    } catch (error) {
+      if (error === "UNMOUNTED") return;
+
+      const { message, errors } = communityErrorSerializer(error);
+
+      if (message) {
+        this.setGlobalError(message);
+      }
+
+      if (errors) {
+        errors.forEach(({ field, messages }) => setFieldError(field, messages[0]));
+      }
+    }
+    this.setIsSavedState(true);
+
+    const savedStateTimer = setTimeout(() => {
```

Comment:
We don't need it for the functionality to work, the user just won't know if the save was successful or not (i.e. no feedback. which is not great). I don't know if we could maybe add an `onChange` event to the `RichInputField` component in react-invenio-forms and then solve it like we do in "Privileges"?

Reply:
I am not sure I understand the reason for using timer, we have the response from backend, which we can use to communicate the state to the user, alongside with the refreshed values. Unless I am missing something

Reply:
I will look more into it...

Reply:
We can use the state from the backend to show a saved state for the button, but the problem is that it never resets. I.e. it will show "saved" even when the user changes the text again. In the "privileges" form, the saved-state is reset on `onChange` on the fields, but there is no `onChange` event on the `RichInputField`, so there is no way to reset the state in this form. That's why I put the timer as a hack just to give some feedback to the user. On the "profile" page, the entire page reloads, so I guess we can do that instead for this page as well.